R/utils.R

```diff
@@ -403,6 +403,57 @@ assert_valid_callbacks <- function(output, params, func) {
     stop(sprintf("The callback method requires that one or more properly formatted inputs are passed."), call. = FALSE)
   }
 
+  # Verify that 'input', 'state' and 'output' parameters only contain 'Wildcard' keywords if they are JSON formatted ids for pattern matching callbacks
+  valid_wildcard_inputs <- sapply(inputs, function(x) {
+    if (grepl("ALL|MATCH|ALLSMALLER", x$id)) {
```

Comment:
(as we discussed offline, the concern here is that we might throw errors if a developer uses a valid name that happens to be a superstring of `ALL`, `MATCH`, or `ALLSMALLER`; it's a bit contrived, but easily avoided, so might as well address it either by ensuring `inputs` are screened before validation or via exact matching on the reserved words only -- as mentioned above)

Reply:
After talking things over with @alexcjohnson, it became evident that I glossed over an important distinction relative to Dash for Julia and Python here. (Sorry about that.) In both of those languages, `ALL`, `MATCH` and `ALLSMALLER` are symbols, rather than strings. In R, I think we could achieve the same effect by doing

```r
ALL <- as.symbol("ALL")
```
which is then convertible to a string (for serializing into JSON) via

```r
dash::to_JSON(as.character(ALL))
```

to yield

```
["ALL"]
```

I'll make the needed changes this morning and push to this branch, I'm just about done, about to test with your sample application.

Reply:
As we talked about earlier, I added your changes in their own commit and updated the callback validation logic to reflect the move to using `symbols` instead of character strings. Updated in 18192cd.